### PR TITLE
En 674 pages endpoints

### PR DESCRIPTION
**Pull request by paddeo**

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

File: engine/src/main/java/org/entando/entando/aps/system/services/page/PageService.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        } catch (ApsSystemException e) {
+            logger.error("Error updating page {}", pageCode, e);
+            throw new RestServerError("error in update page", e);
+        }
+    }
+
+    private IPage createPage(PageRequest pageRequest) {
+        Page page = new Page();
+        page.setCode(pageRequest.getCode());
+        page.setShowable(pageRequest.isDisplayedInMenu());//TODO
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/page/PageService.java]: Is this TODO done?

File: engine/src/main/java/org/entando/entando/aps/system/services/page/PageService.java [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+        PageModel model = this.getPageModelManager().getPageModel(pageRequest.getPageModel());
+        page.setModel(model);
+        page.setCharset(pageRequest.getCharset());
+        page.setMimeType(pageRequest.getContentType());
+        page.setParentCode(pageRequest.getParentCode());
+//        page.setSeo(false);//TODO
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/page/PageService.java]: Is this TODO done?

File: engine/src/main/java/org/entando/entando/aps/system/services/page/PageService.java [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+            IPage parent = this.getPageManager().getDraftPage(pageRequest.getParentCode());
+            page.setParent(parent);
+        }
+        return page;
+    }
+
+    private IPage updatePage(IPage oldPage, PageRequest pageRequest) {
+        Page page = new Page();
+        page.setCode(pageRequest.getCode());
+        page.setShowable(pageRequest.isDisplayedInMenu());//TODO
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/page/PageService.java]: Is this TODO done?

File: engine/src/main/java/org/entando/entando/aps/system/services/page/PageService.java [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+        if (oldPage.getModel().getCode().equals(pageRequest.getPageModel())) {
+            PageModel model = this.getPageModelManager().getPageModel(pageRequest.getPageModel());
+            page.setModel(model);
+        }
+        page.setCharset(pageRequest.getCharset());
+        page.setMimeType(pageRequest.getContentType());
+        page.setParentCode(pageRequest.getParentCode());
+//        page.setSeo(false);//TODO
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/page/PageService.java]: Is this TODO done?

File: engine/src/main/java/org/entando/entando/aps/system/services/page/model/PageDto.java [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+    private String ownerGroup;
+    private List<String> joinGroups = new ArrayList<>();
+    private int position;
+
+    public PageDto() {
+    }
+
+    public PageDto(IPage page) {
+        this.setCode(page.getCode());
+        this.setStatus(page.isOnline() ? "online" : "draft");//TODO
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/page/model/PageDto.java]: Is this TODO done?

File: engine/src/main/java/org/entando/entando/aps/system/services/page/model/PageDto.java [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+        this.setDisplayedInMenu(page.isShowable());//TODO
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/page/model/PageDto.java]: Is this TODO done?

Comment (paddeo) [reply engine/src/main/java/org/entando/entando/aps/system/services/page/model/PageDto.java]: all TODOs done and removed

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Pull request merged**